assets/src/edit-story/components/panels/stylePreset/presets.js

```diff
@@ -0,0 +1,279 @@
+  height: ${PRESET_HEIGHT}px;
+`;
+
+const StyleButtonWrapper = styled.div`
+  flex-basis: ${100 / STYLES_PER_ROW}%;
+  height: ${PRESET_HEIGHT}px;
+`;
+
+const PresetGroupLabel = styled.div`
+  color: ${({ theme }) => theme.colors.fg.v1};
+  font-size: 10px;
+  line-height: 12px;
+  text-transform: uppercase;
+  padding: 6px 0;
+`;
+
+const TextWrapper = styled.span`
+  text-align: left;
+  line-height: 1.3;
+`;
+
+const HighLight = styled.span`
+  padding: 0 2px;
+  ${({ background }) => generatePatternStyles(background)}
+  box-decoration-break: clone;
+`;
+
+function Presets({
+  stylePresets,
+  getEventHandlers,
+  isEditMode,
+  isText,
+  textContent = 'Text',
+}) {
+  const [activeColorIndex, setActiveColorIndex] = useState(null);
+  const [activeStyleIndex, setActiveStyleIndex] = useState(null);
+  const colorsRef = useRef(null);
+  const stylesRef = useRef(null);
+  const { fillColors, textColors, textStyles } = stylePresets;
+
+  const getIndexDiff = (key, rowLength) => {
+    switch (key) {
+      case 'ArrowUp':
+        return -rowLength;
+      case 'ArrowDown':
+        return COLORS_PER_ROW;
+      case 'ArrowLeft':
+        return -1;
+      case 'ArrowRight':
+        return 1;
+      default:
+        return 0;
+    }
+  };
+
+  useLayoutEffect(() => {
+    if (colorsRef.current && null !== activeColorIndex) {
+      colorsRef.current.querySelector('[tabindex="0"]').focus();
+    }
+  }, [activeColorIndex]);
+
+  useLayoutEffect(() => {
+    if (stylesRef.current && null !== activeStyleIndex) {
+      stylesRef.current.querySelector('[tabindex="0"]').focus();
+    }
+  }, [activeStyleIndex]);
+
+  useKeyDownEffect(
+    colorsRef,
+    { key: ['up', 'down', 'left', 'right'], shift: true },
+    ({ key }) => {
+      const maxIndex = isText ? textColors.length - 1 : fillColors.length - 1;
+      if (colorsRef.current) {
+        const diff = getIndexDiff(key, COLORS_PER_ROW);
+        const newIndex = Math.max(
+          0,
+          Math.min(maxIndex, (activeColorIndex ?? 0) + diff)
+        );
+        setActiveColorIndex(newIndex);
+      }
+    },
+    [activeColorIndex]
+  );
+
+  useKeyDownEffect(
+    stylesRef,
+    { key: ['up', 'down', 'left', 'right'], shift: true },
+    ({ key }) => {
+      if (stylesRef.current) {
+        const diff = getIndexDiff(key, STYLES_PER_ROW);
+        const newIndex = Math.max(
+          0,
+          Math.min(textStyles.length - 1, (activeStyleIndex ?? 0) + diff)
+        );
+        setActiveStyleIndex(newIndex);
+      }
+    },
+    [activeStyleIndex]
+  );
+
+  const getStylePresetText = (preset) => {
+    const isHighLight =
+      preset.backgroundTextMode === BACKGROUND_TEXT_MODE.HIGHLIGHT;
+    // @todo Confirm text content usage.
+    const text = textContent.substring(0, 8) + '...';
+    return (
+      <TextWrapper>
+        {isHighLight ? (
+          <HighLight background={preset.backgroundColor}>{text}</HighLight>
+        ) : (
+          text
```

Comment:
Shouldn't `BACKGROUND_TEXT_MODE.FILL` text have its background color shown too?

Reply:
That's added directly to the background on the preset, the highlight is not covering the full preset area whereas the `FILL` background is.